proposals/001-file-format-api.md

```diff
@@ -0,0 +1,608 @@
+`Text` | text/plain
+`Tiff` | image/tiff
+`Ttf` | font/ttf
+`Wav` | audio/wav
+`WebmAudio` | audio/webm
+`WebmVideo` | video/webm
+`Webp` | image/webp
+`WebPub` | application/webpub+zip
+`WebPubManifest` | application/webpub+json
+`Woff` | font/woff
+`Woff2` | font/woff2
+`Xhtml` | application/xhtml+xml
+`Xml` | application/xml
+`Zip` | application/zip
+
+
+### `Format` Class
+
+Instead of using directly a media type, we use a `Format` data class to identify a file format for several reasons:
+
+* A format such as CBZ can be identified by several media types.
+* It's a convenient way to pair a media type and its name and file extension.
+
+#### Constructor
+
+Note: all of the constructor parameters are exposed as read-only public properties.
+
+* `Format(mediaType: MediaType, name: String, fileExtension: String)`
+  * `mediaType: MediaType`
+    * The canonical media type that identifies the best (most officially) this format.
+    * This will be stored in the reading app's database, for example.
+  * `name: String`
+    * A human readable name describing the format.
+    * Can be presented to the user.
+  * `fileExtension: String`
+    * The default file extension to use for this format.
+
+#### Helpers
+
+* `isManifest`, `isOpds`, `isHtml`, `isBitmap`...
+  * Aliases to `mediaType.isX`
+
+#### Constants
+
+* `defaultSniffers: List<Sniffer>`
+  * The default sniffers provided by Readium 2 to guess a `Format`.
+  * This list is writable, to allow reading apps to register additional sniffers globally, or remove sniffers.
+
+Formats used by Readium are represented as static constants (singletons) on `Format`, for convenience.
+Reading apps are welcome to extend the static constants with additional formats.
+
+Constant | Name | Extension | Media Type
+-------- | ---- | --------- | ----------
+`Audiobook` | Audiobook | audiobook | application/audiobook+zip
+`AudiobookManifest` | Audiobook | json | application/audiobook+json
+`Cbz` | Comic Book Archive | cbz | application/vnd.comicbook+zip
+`DiViNa` | Digital Visual Narratives | divina | application/divina+zip
+`DiViNaManifest` | Digital Visual Narratives | json | application/divina+json
+`Epub` | EPUB | epub | application/epub+zip
+`Html` | HTML | html | text/html
+`Opds1Feed` | OPDS | atom | application/atom+xml;profile=opds-catalog
+`Opds1Entry` | OPDS | atom | application/atom+xml;type=entry;profile=opds-catalog
+`Opds2Feed` | OPDS | json | application/opds+json
+`Opds2Publication` | OPDS | json | application/opds-publication+json
+`LcpProtectedAudiobook` | LCP Protected Audiobook | lcpa | application/audiobook+lcp
+`LcpProtectedPdf` | LCP Protected PDF | lcpdf | application/pdf+lcp
+`LcpLicense` | LCP License | lcpl | application/vnd.readium.lcp.license.v1.0+json
+`Pdf` | PDF | pdf | application/pdf
+`WebPub` | Web Publication | webpub | application/webpub+zip
+`WebPubManifest` | Web Publication | json | application/webpub+json
+`Zab` | Zipped Audio Book | zab | application/vnd.zab+zip
```

Comment:
I couldn't find any media type online for ZAB, so I created this one from scratch: `application/vnd.zab+zip`

Reply:
I used the `x.` facet instead.